### DISCUSSION TITLE
Is there a way to export .vrm models?

### DISCUSSION THREAD

**Discussion by ALABOET** [pixiv/three-vrm#1114] [Help]
Maybe this is a stupid question so sorry in advance, i am new to three js and 3d as a whole.
So I am trying to find a way to export the model. And even though i found a "way" to export it through GLTFExporter, the model is all gray with no textures.  So i was thinking, maybe there is another way to do it? Thank you!

**Comment by 0b5vr** [comment]
I have tried to create a VRM plugin for GLTFExporter (like how three-vrm does in GLTFLoader) once in the past, but it could not meet the point of being usable.

**Reply by ALABOET** [reply]
may i ask why you tried to do that? Is it because there didnt / doesnt seem to be an alternative to exporting a fully working model? As i said, GLTFExporter doesnt seem to work properly for me :) Did u find any solution?

**Reply by ALABOET** [reply]
i was just thinking, maybe it is possible to mimic what happens in Blender when we export a vrm model, to Three js? I guess that would be tough since its Python there but i really dont see any other way.

**Reply by 0b5vr** [reply]
We need to support [`MToonMaterial`](https://pixiv.github.io/three-vrm/packages/three-vrm/docs/classes/MToonMaterial.html) in GLTFExporter to support your broken exports of materials and textures; that's why your exports have broken I believe.
